src/Microsoft.AspNet.Mvc.Razor/VirtualPathRazorPageFactory.cs

```diff
@@ -15,14 +15,20 @@ namespace Microsoft.AspNet.Mvc.Razor
     public class VirtualPathRazorPageFactory : IRazorPageFactory
     {
         private readonly IServiceProvider _serviceProvider;
-        private readonly ICompilerCache _compilerCache;
+        private readonly ICompilerCacheProvider _compilerCacheProvider;
         private IRazorCompilationService _razorcompilationService;
 
-        public VirtualPathRazorPageFactory(IServiceProvider serviceProvider,
-                                           ICompilerCache compilerCache)
+        /// <summary>
+        /// Initializes a new instance of <see cref="VirtualPathRazorPageFactory"/>.
+        /// </summary>
+        /// <param name="serviceProvider">The request specific <see cref="IServiceProvider"/>.</param>
+        /// <param name="compilerCacheProvider">The <see cref="ICompilerCacheProvider"/>.</param>
+        public VirtualPathRazorPageFactory(
+            IServiceProvider serviceProvider,
+            ICompilerCacheProvider compilerCacheProvider)
         {
             _serviceProvider = serviceProvider;
```

Comment:
Slightly related but I don't think https://github.com/aspnet/Mvc/pull/3004/files#diff-88150cf385250544dc6321e6ef7c28fdR42 is true. If not, I'll clean it up as a follow up to this.